cmd/notice-file-generator/dependency.go

```diff
@@ -346,12 +347,41 @@ func PopulateGoDependencies(config *Config) ([]Dependency, error) {
 	return dependencies, nil
 }
 
+func PopulatePythonDependencies(config *Config) ([]Dependency, error) {
+	dependencies := []Dependency{}
+	inFile, err := os.Open(config.Path + "/Pipfile")
+	if err != nil {
+		log.Fatalf("Invalid pipfile. %v", err)
+	}
+	defer inFile.Close()
+	scanner := bufio.NewScanner(inFile)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "# Repo:") {
+			repoURL := strings.Trim(strings.TrimPrefix(line, "# Repo:"), " ")
+			if strings.HasPrefix(repoURL, "https://github.com") {
+				dependencies = append(dependencies, Dependency{
+					Name:     path.Base(repoURL),
+					FullName: strings.TrimPrefix(repoURL, "https://github.com/"),
+					Repository: DependencyRepository{
+						Type: "https",
+						URL:  repoURL,
+					},
+				})
+			}
+		}
+	}
+	return dependencies, nil
+}
+
 func PopulateDependencies(config *Config) ([]Dependency, error) {
 
 	if config.IsJsRepo() {
```

Comment:
minor: what if you make it a bit more generic and avoid the multiple for loops in each `Is<Lang>Repo` function? may the for loop can be here and use a switch/case? need to think a bit more but I believe is possible to avoid the for loop in each function.

Reply:
Yes that function needs refactoring. Also this will reduce the needed work to support multi project at same repo(most of the plugins have go and npm projects at the same repository).

Reply:
Introduced a RepoType enum, not to loop it everywhere. It will be set at the configuration initialization and that property will be used at everywhere.